crates/sncast/src/helpers/interactive.rs

```diff
@@ -1,64 +1,69 @@
 use crate::helpers::config::get_global_config_path;
-use anyhow::{anyhow, Context, Result};
+use anyhow::{Context, Result};
 use camino::Utf8PathBuf;
 use configuration::search_config_upwards_relative_to;
 use dialoguer::theme::ColorfulTheme;
 use dialoguer::Select;
 use std::env::current_dir;
+use std::fmt::Display;
 use std::fs;
 use toml_edit::{DocumentMut, Item, Table, Value};
 
+enum PromptSelection {
+    GlobalDefault(String, Utf8PathBuf),
+    LocalDefault(String, Utf8PathBuf),
+    No,
+}
+
+impl Display for PromptSelection {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PromptSelection::LocalDefault(s, _) | PromptSelection::GlobalDefault(s, _) => {
+                write!(f, "{s}")
+            }
+            PromptSelection::No => write!(f, "No"),
+        }
+    }
+}
+
 pub fn prompt_to_add_account_as_default(account: &str) -> Result<()> {
     let mut options = Vec::new();
 
-    if let Some(option) = get_global_config_path().ok().map(|global_path| {
-        format!(
-            "Yes, global default account ({}).",
-            to_tilde_path(&global_path)
-        )
-    }) {
-        options.push(option);
+    if let Ok(global_path) = get_global_config_path() {
+        options.push(PromptSelection::GlobalDefault(
+            format!("Yes, global default ({})", to_tilde_path(&global_path)),
+            global_path.clone(),
+        ));
     }
 
-    if let Some(option) = current_dir()
+    if let Some(local_path) = current_dir()
         .ok()
         .and_then(|current_path| Utf8PathBuf::from_path_buf(current_path.clone()).ok())
         .and_then(|current_path_utf8| search_config_upwards_relative_to(&current_path_utf8).ok())
-        .map(|local_path| {
-            format!(
-                "Yes, local default account ({}).",
-                to_tilde_path(&local_path)
-            )
-        })
     {
-        options.push(option);
+        options.push(PromptSelection::LocalDefault(
+            format!("Yes, local default ({})", to_tilde_path(&local_path)),
+            local_path.clone(),
+        ));
```

Comment:
```suggestion
        options.push(PromptSelection::LocalDefault(local_path.clone()));
```

Also check if and where `Clone` is necessary